tiledb/libtiledb.pyx

```diff
@@ -4444,7 +4444,7 @@ cdef class Query(object):
     def __getitem__(self, object selection):
         return self.array.subarray(selection,
                                    attrs=self.attrs,
-                                   coords=self.coords,
+                                   coords=self.coords if self.coords else self.dims,
```

Comment:
I believe this logic needs to be inverted so that `dims` takes precedence.

Reply:
What would be a case where this would give an unexpected result?

Reply:
`coords=True, dims=["a"]`

Reply:
It should error out in `__init__` before then with `Cannot pass both dims and coords=True to Query`

Reply:
Good point, ok.